### PR TITLE
recipes-web: Update iotivity-node and iot-rest-api-server to latest

### PR DESCRIPTION
**Pull request by nagineni**
Pull in the iotivity-node version 1.2.0-2. 

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>

### CODE REVIEW

**Comment by nagineni** [comment]
@mikko @poussa

**Comment by poussa** [comment]
+1

File: recipes-web/iot-rest-api-server/iot-rest-api-server.bb [hunk 1]
```diff
@@ -27,8 +27,7 @@ SYSTEMD_AUTO_ENABLE_${PN} = "disable"
 USERADD_PACKAGES = "${PN}"
 GROUPADD_PARAM_${PN} = "-r restful"
 USERADD_PARAM_${PN} = "\
---system --home ${localstatedir}/lib/empty \
---no-create-home --shell /bin/false \
+--system --create-home --shell /bin/false \
```
Comment (mythi) [inline recipes-web/iot-rest-api-server/iot-rest-api-server.bb:30]: the commit message says iotivity-node installs to home but did you mean to say iot-rest-api-server?

with useradd-staticids.bblass used in Ostro, the home dir is picked from ostro-passwd and is defined as ```/var/lib/empty```. Was the problem that iot-rest-api-server was not able to create that dir but now that it gets created here it's OK? Perhaps explain the problem in the commit message.

Comment (nagineni) [reply recipes-web/iot-rest-api-server/iot-rest-api-server.bb:30]: @mythi Updated the commit message.

Issue was that, iotivity-node instantiated by rest api server was failing to create configuration files under ${HOME}/.iotivity-node directory when the server started under non-root user (i.e restful). The reason for that was the restful user was created without a home directory. So, I changed --no-create-home option to --create-home to create user's home directory.

**Comment by mythi** [comment]
On Fri, Feb 3, 2017 at 11:15 AM, Sudarsana Nagineni  wrote:


> @mythi <https://github.com/mythi> Updated the commit message.
>
> Issue was that, iotivity-node instantiated by rest api server was failing
> to create configuration files under ${HOME}/.iotivity-node directory when
> the server started under non-root user (i.e restful). The reason for that
> was the restful user was created without a home directory. So, I changed
> --no-create-home option to --create-home to create user's home directory.
>
>
You should be getting /var/lib/empty with this change.

**Comment by nagineni** [comment]
> You should be getting /var/lib/empty with this change.

Yes, that is what we need for now to solve the issue. I will change the dir to /home/restful in ostro-passwd in a separate PR later.

**Comment by mythi** [comment]
+1

**Pull request merged**